src/users/userSelectors.js

```diff
@@ -33,14 +33,24 @@ export const getAllActiveUsers = createSelector(getUsers, allUsers =>
   allUsers.filter(user => user.isActive),
 );
 
+export const getAllActiveUsersWithStatus = createSelector(
+  getAllActiveUsers,
+  getPresence,
+  (allUsers, presence) =>
+    allUsers.map(user => ({
+      ...user,
```

Comment:
That will have very bad performance, though from code expressiveness is good.
On each iteration, you will be cloning the whole object.
Look at `reduce()`